File: Distancia2puntosC3d/Distancia2puntosC/PuntoXY.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Distancia2puntosC
{
    class PuntoXY:Punto_X 
    {
        protected double y;
        public double Y
        {
            get { return y; }
            set { y = value; }
        }
        public PuntoXY():base()
        {
            y = 0;
        }
        public PuntoXY(double X, double Y):base (X)
        {
            this.y = Y;
        }

        public double  CalcularDistancia(PuntoXY a)
        {
            return Math.Sqrt(Math.Pow((x - a.X), 2) + Math.Pow((y - a.Y), 2));
        }

        public override string ToString()
        {
            return x + "," + y;
        }
    }
   
}
   

File: Distancia2PuntosVB3d/Distancia2PuntosVB/PuntoXY.vb
Public Class PuntoXY
    Inherits PuntoX

    Protected y As Double

    Public Property _Y As Double
        Get
            Return y
        End Get
        Set(ByVal value As Double)
            y = value
        End Set
    End Property

    Public Sub New()
        MyBase.New()
        y = 0
    End Sub

    Public Sub New(ByVal _X As Double, ByVal _Y As Double)
        MyBase.New(_X)
        Me.y = _Y
    End Sub

    Public Function CalcularDistancia(ByVal a As PuntoXY) As Double
        Return Math.Sqrt(Math.Pow((x - a._X), 2) + Math.Pow((y - a._Y), 2))
    End Function

    Public Overrides Function ToString() As String
        Return x & "," + y
    End Function


End Class
